src/python/pants/backend/python/macros/python_requirements.py

```diff
@@ -110,9 +152,16 @@ async def generate_from_python_requirement(
             description_of_origin=f"{generator}'s field `{PythonRequirementsSourceField.alias}`",
         ),
     )
-    requirements = parse_requirements_file(
-        digest_contents[0].content.decode(), rel_path=requirements_full_path
-    )
+    if requirements_rel_path.endswith("toml"):
```

Comment:
This could also be `.endswith("pyproject.toml")` but this felt like a good enough check to me. I don't think it should be `== "pyproject.toml"` because that would fail for subdirectories, I think.

Reply:
Ok. Then use this: `os.path.basename(x) == "pyproject.toml"` - that's really the right thing to do.

Reply:
Done @jsirois! Thanks for the suggestion 😄